src/jdk.crypto.ec/share/classes/sun/security/ec/ECDHKeyAgreement.java

```diff
@@ -167,15 +204,13 @@ private static void validate(ECOperations ops, ECPublicKey key) {
         if ((privateKey == null) || (publicKey == null)) {
             throw new IllegalStateException("Not initialized correctly");
         }
+
         byte[] result;
-        Optional<byte[]> resultOpt = deriveKeyImpl(privateKey, publicKey);
-        if (resultOpt.isEmpty()) {
-            NamedCurve nc = CurveDB.lookup(publicKey.getParams());
-            throw new IllegalStateException(
-                new InvalidAlgorithmParameterException("Curve not supported: " +
-                    (nc != null ? nc.toString() : "unknown")));
+        try {
+            result = deriveKeyImpl(privateKey, privateKeyOps, publicKey);
+        } catch (Exception e) {
```

Comment:
Why such a broad exception catch here?  It looks like deriveKeyImpl is only explicitly throwing IKE.  Are there other unchecked exceptions that you're trying to snag here that I'm missing in the deriveKeyImpl below?

Reply:
Just being cautious and wrapping anything.  Maybe there will be some exceptions in the math methods that throw an exception.

Reply:
Fair enough.  I'm OK to leave this as-is.